aws-cloudformation/aws-cloudformation-resource-providers-ecr
# add RegistryPolicy resource

*Description of changes:*


* The RegistryPolicy schema definition.
* CRUD operations for Uluru support (not supporting List)
* Unit Tests for the CRUD operations.

*Example CloudFormation templates*
JSON:
~~~
{
    "Resources": {
       "TestRegistryPolicy": {
              "Type": "AWS::ECR::RegistryPolicy",
              "Properties": {
                "PolicyText": {
                        "Version":"2012-10-17",
                        "Statement":[
                            {
                                "Sid":"ReplicationAccessCrossAccount",
                                "Effect":"Allow",
                                "Principal":{
                                    "AWS":"arn:aws:iam::123456789:root"
                                },
                                "Action":[
                                    "ecr:CreateRepository",
                                    "ecr:ReplicateImage"
                                ],
                                "Resource": "arn:aws:ecr:us-west-2:123412341234:repository/*"
                            }
                        ]
                    }
              }
        }
    }    
}
~~~

YAML:
~~~
Resources:
  TestRegistryPolicy:
    Type: 'AWS::ECR::RegistryPolicy'
    Properties:
      PolicyText:
        Version: 2012-10-17
        Statement:
          - Sid: UpdatedRegistryPolicy
            Effect: Allow
            Principal:
              AWS: 'arn:aws:iam::123456789:root'
            Action:
              - 'ecr:CreateRepository'
              - 'ecr:ReplicateImage'
            Resource: 'arn:aws:ecr:us-west-2:123412341234:repository/*'
~~~

*Testing*
* mvn package
* cfn test
* cfn submit
* manual testing
  * created CFN stack with example template above and verified RegistryPolicy was applied to my registry.
  * deleted stack after creation and verified registry policy deleted from registry
  * created stack then updated with new template, verifying changes seen in ECR registry policy



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
